lib/graphql/schema.rb

```diff
@@ -349,13 +349,9 @@ def lazy?(obj)
       !!lazy_method_name(obj)
     end
 
-    # @return [Array<#call>] Middlewares suitable for MiddlewareChain, applied to fields during execution
+    # @return [MiddlewareChain] MiddlewareChain which is applied to fields during execution
     def middleware
-      @middleware ||= if @rescue_middleware
-        [@rescue_middleware].concat(@user_middleware)
-      else
-        @user_middleware
-      end
+      @user_middleware
```

Comment:
Is this code testing that behavior? 

https://github.com/rmosolgo/graphql-ruby/blob/master/spec/graphql/schema_spec.rb#L302:
```ruby
it "copies internal state" do
  schema_2 = schema.dup
  refute schema_2.types.equal?(schema.types)

  refute schema_2.instrumenters.equal?(schema.instrumenters)
  assert_equal schema_2.instrumenters, schema.instrumenters

  refute schema_2.middleware.equal?(schema.middleware)
  assert_equal schema_2.middleware, schema.middleware
end
```